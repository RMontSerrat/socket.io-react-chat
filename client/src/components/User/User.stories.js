import React from 'react';
import { storiesOf } from '@storybook/react';
import User from './User';

storiesOf('User', module)
  .add('default', () => (
    <div style={{ padding: '20px' }}>
      <User>Romulo</User>
    </div>
  ));
