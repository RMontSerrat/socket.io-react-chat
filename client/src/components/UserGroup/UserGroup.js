import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const UserGroupContainer = styled.div`
  border-radius: ${props => props.theme.borderRadius};
  background: ${props => props.theme.colorWhite};
  padding: ${props => props.theme.padding};
  color: ${props => props.theme.colorPrimary};
`

const Users = styled.div`
  display: flex;
  flex-direction: column;
  @media (max-width: 600px) {
    flex-direction: row;
    > div {
      padding-right: 10px;
    }
  }
`
const Title = styled.h2`
  margin: 0;
  padding-bottom: 10px;
`
const UserGroup = ({ children }) => (
  <UserGroupContainer>
    <Title>Usu√°rios</Title>
    <Users>
      {children}
    </Users>
  </UserGroupContainer>
)

UserGroup.propTypes = {
  children: PropTypes.node,
}

export default UserGroup;