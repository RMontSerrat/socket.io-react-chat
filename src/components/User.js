import React from 'react';
import styled from 'styled-components';
import UserIcon from '../icons/User';

const UserContainer = styled.div`
  padding: 5px 0;
  color: ${props => props.theme.colorPrimary};
  font-family: ${props => props.theme.fontFamily};
  display: flex;
  font-size: 14px;
  align-items: center;
`
const Text = styled.span`
  padding-left: 10px;
`

const User = ({ children }) => 
  <UserContainer>
    <UserIcon />
    <Text>{children}</Text>
  </UserContainer>

export default User;