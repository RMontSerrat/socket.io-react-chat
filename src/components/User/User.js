import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import UserIcon from '../../icons/User';

const UserContainer = styled.div`
  padding: 5px 0;
  color: ${props => props.theme.color.primary};
  font-family: ${props => props.theme.font.primary};
  display: flex;
  font-size: 14px;
  align-items: center;
`
const Text = styled.span`
  padding-left: 10px;
`

const User = ({ children }) => 
  <UserContainer>
    <UserIcon />
    <Text>{children}</Text>
  </UserContainer>

User.propTypes = {
  children: PropTypes.node,
}

export default User;