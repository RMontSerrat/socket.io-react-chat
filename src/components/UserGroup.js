import React from 'react';
import styled from 'styled-components';

const UserGroupContainer = styled.div`
  border-radius: ${props => props.theme.borderRadius};
  background: ${props => props.theme.colorWhite};
  padding: 15px;
  color: ${props => props.theme.colorPrimary};
`
const Title = styled.h2`
  font-family: ${props => props.theme.fontFamily};
`
const UserGroup = ({ children }) => (
  <UserGroupContainer>
    <Title>Usu√°rios</Title>
    {children}
  </UserGroupContainer>
)
export default UserGroup;