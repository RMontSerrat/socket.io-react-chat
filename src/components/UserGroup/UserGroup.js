import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const UserGroupContainer = styled.div`
  border-radius: ${props => props.theme.borderRadius};
  background: ${props => props.theme.colorWhite};
  padding: ${props => props.theme.padding};
  color: ${props => props.theme.colorPrimary};
`
const Title = styled.h2`
  font-family: ${props => props.theme.fontFamily};
  margin: 0;
  padding-bottom: 10px;
`
const UserGroup = ({ children }) => (
  <UserGroupContainer>
    <Title>Usu√°rios</Title>
    {children}
  </UserGroupContainer>
)

UserGroup.propTypes = {
  children: PropTypes.node,
}

export default UserGroup;