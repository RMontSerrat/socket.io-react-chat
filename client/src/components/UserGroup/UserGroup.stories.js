import React from 'react';
import { storiesOf } from '@storybook/react';
import UserGroup from './UserGroup';
import User from '../User';

storiesOf('UserGroup', module)
  .add('default', () => (
    <div style={{ backgroundColor: '#025a3f', padding: '20px' }}>
      <UserGroup>
        <User>Romulo</User>
        <User>Jorge</User>
        <User>ZÃ©</User>
      </UserGroup>
    </div>
  ));
