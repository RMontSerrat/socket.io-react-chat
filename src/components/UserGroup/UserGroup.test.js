import React from "react";
import { render, cleanup } from "react-testing-library";
import UserGroup from "./UserGroup";
import User from '../User';

afterEach(cleanup);
describe('<UserGroup />', () => {
  it("should render", () => {
    const { asFragment } = render(
      <UserGroup>
        <User>Romulo</User>
      </UserGroup>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});