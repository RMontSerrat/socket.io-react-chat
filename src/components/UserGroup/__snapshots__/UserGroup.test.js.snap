// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserGroup /> should render 1`] = `
<DocumentFragment>
  .c0 {
  margin: 0;
  padding-bottom: 10px;
}

.c2 {
  width: px;
  height: px;
}

.c1 {
  padding: 5px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 14px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  padding-left: 10px;
}

<div
    class=""
  >
    <h2
      class="c0"
    >
      Usu√°rios
    </h2>
    <div
      class="c1"
    >
      <svg
        class="c2"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
          fill="currentColor"
        />
      </svg>
      <span
        class="c3"
      >
        Romulo
      </span>
    </div>
  </div>
</DocumentFragment>
`;
